showdependencygraph.cgi

```diff
@@ -167,166 +176,52 @@ else {
-  ThrowUserError("webdot_too_large");
-}
-
-my $webdotbase = Bugzilla->params->{'webdotbase'};
-
-if ($webdotbase =~ /^https?:/) {
-  $webdotbase =~ s/%(?:sslbase|urlbase)%/Bugzilla->localconfig->urlbase/eg;
-  my $url = $webdotbase . $filename;
-  $vars->{'image_url'} = $url . ".gif";
-  $vars->{'map_url'}   = $url . ".map";
+my $urlbase = Bugzilla->localconfig->urlbase;
+foreach my $k (keys %seen) {
+  $graph .= qq{click $k "${urlbase}show_bug.cgi?id=$k"\n};
 }
-else {
-  # Local dot installation
-  my $image_data;
-  open(my $pipe, '-|', $webdotbase, '-Tpng', $filename) || ThrowCodeError('webdot_error');
-  binmode $pipe;
-  while (my $dot_data = <$pipe>) {
-    $image_data .= $dot_data;
-  }
-  close $pipe || ThrowCodeError('webdot_error');
-
-  $vars->{'image_data'} = encode_base64($image_data);
-
-  # Then, generate a imagemap datafile that contains the corner data
-  # for drawn bug objects. Pass it on to $CreateImagemap that
-  # turns this monster into HTML.
 
-  my ($mapfh, $mapfilename)
-    = File::Temp::tempfile("XXXXXXXXXX", SUFFIX => '.map', DIR => $webdotdir);
+$vars->{'graph_data'} = $graph;
 
-  chmod Bugzilla::Install::Filesystem::WS_SERVE, $mapfilename
-    or warn install_string('chmod_failed', {path => $mapfilename, error => $!});
-
-  binmode $mapfh;
-  open(DOT, '-|', "\"$webdotbase\" -Tismap $filename");
-  binmode DOT;
-  print $mapfh $_ while <DOT>;
-  close DOT;
-  close $mapfh;
-
-  $vars->{'image_map'} = $CreateImagemap->($mapfilename);
-}
-
-# Cleanup any old .dot files created from previous runs.
-my $since = time() - 24 * 60 * 60;
-
-# Can't use glob, since even calling that fails taint checks for Perl < 5.6
-opendir(DIR, $webdotdir);
-my @files = grep { /\.dot$|\.png$|\.map$/ && -f "$webdotdir/$_" } readdir(DIR);
-closedir DIR;
-foreach my $f (@files) {
-  $f = "$webdotdir/$f";
-
-  # Here we are deleting all old files. All entries are from the
-  # $webdot directory. Since we're deleting the file (not following
-  # symlinks), this can't escape to delete anything it shouldn't
-  # (unless someone moves the location of $webdotdir, of course)
-  my $mtime = file_mod_time($f);
-  if ($mtime && $mtime < $since) {
-    unlink $f;
-  }
+if ($bug_count > MAX_DEP_GRAPH_BUGS) {
+  ThrowUserError('dep_graph_too_large');
 }
 
 # Make sure we only include valid integers (protects us from XSS attacks).
-my @bugs = grep(detaint_natural($_), split(/[\s,]+/, $cgi->param('id')));
-$vars->{'bug_id'}        = join(', ', @bugs);
-$vars->{'multiple_bugs'} = ($cgi->param('id') =~ /[ ,]/);
-$vars->{'display'}       = $display;
-$vars->{'rankdir'}       = $rankdir;
-$vars->{'showsummary'}   = $cgi->param('showsummary');
+my @bugs = grep { detaint_natural($_) } split /[\s,]+/, $cgi->param('id');
+$vars->{'bug_id'}      = join ', ', @bugs;
+$vars->{'display'}     = $display;
+$vars->{'rankdir'}     = $rankdir;
+$vars->{'showsummary'} = $cgi->param('showsummary');
+$vars->{'debug'}       = ($cgi->param('debug') ? 1 : 0);
```

Comment:
Do we still need `debug`?

Reply:
Yeah I think so. I feel like there may be more corner cases later on that some characters, etc. in bug summaries will break the graph markup text and will need to be debugged. Having the debug=1 available, I can just ask people to paste the graph text to a bug report and I can then debug it locally.